tests/databases/constructed_molecule/mongo_db/test_constructed_molecule_update.py

```diff
@@ -0,0 +1,426 @@
+            DatabaseEntry(
+                **to_hashable_matrix(
+                    json=json['buildingBlocks'][0]['matrix'],
+                )
+            ): 1,
+            DatabaseEntry(
+                **to_hashable_constructed_molecule(
+                    json=json['constructedMolecule'],
+                )
+            ): 1,
+        }),
+    )
+
+    polymer2 = polymer.with_position_matrix(
+        position_matrix=np.zeros((polymer.get_num_atoms(), 3)),
+    )
+    json2 = jsonizer.to_json(polymer2)
+
+    database.put(polymer2)
+    assert_database_state(
+        state1=get_database_state(database),
+        state2=DatabaseState({
+            DatabaseEntry(**json['molecule']): 1,
+            DatabaseEntry(**to_hashable_matrix(json2['matrix'])): 1,
+            DatabaseEntry(**json['buildingBlocks'][0]['molecule']): 1,
+            DatabaseEntry(
+                **to_hashable_matrix(
+                    json=json['buildingBlocks'][0]['matrix'],
+                )
+            ): 1,
+            DatabaseEntry(
+                **to_hashable_constructed_molecule(
+                    json=json['constructedMolecule'],
+                )
+            ): 1,
+        }),
+    )
+
+
+def test_update_2():
+    """
+    Test that existing entries are updated.
+
+    In this test, your first create two separate entries, using
+    different molecule keys. You then update both at the same time,
+    with a database which uses both molecule keys.
+
+    """
+
+    database_name = '_test_update_2'
+    client = pymongo.MongoClient()
+    client.drop_database(database_name)
+
+    jsonizer1 = stk.ConstructedMoleculeJsonizer(
+        key_makers=(
+            stk.InchiKey(),
+        ),
+    )
+    database1 = stk.ConstructedMoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        jsonizer=jsonizer1,
+    )
+
+    jsonizer2 = stk.ConstructedMoleculeJsonizer(
+        key_makers=(
+            stk.Smiles(),
+        ),
+    )
+    database2 = stk.ConstructedMoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        jsonizer=jsonizer2,
+    )
+
+    jsonizer3 = stk.ConstructedMoleculeJsonizer(
+        key_makers=(
+            stk.InchiKey(),
+            stk.Smiles(),
+        ),
+    )
+    database3 = stk.ConstructedMoleculeMongoDb(
+        mongo_client=client,
+        database=database_name,
+        put_lru_cache_size=0,
+        get_lru_cache_size=0,
+        jsonizer=jsonizer3,
+    )
+
+    molecule = stk.BuildingBlock(
+        smiles='BrCCCBr',
+        functional_groups=[stk.BromoFactory()],
+    ).with_canonical_atom_ordering()
+
+    polymer1 = stk.ConstructedMolecule(
+        topology_graph=stk.polymer.Linear(
+            # Use it as a building block twice, to make sure it is
```

Comment:
This does not make sense - why do you need to do it and is this something the user needs to be aware of?

Reply:
It is purely for testing purposes. If I deposit the polymer, which has a building block twice, I want to make sure it does not create two entries for that building block but just 1.